src/generators/primes.rs

```diff
@@ -2,56 +2,53 @@ extern crate num;
 
 pub struct PrimeSequence<T: num::PrimInt> {
     primes: Vec<T>,
-    n: T,         // 6n±1
-    bottom: bool, // true = 6n-1, false = 6n+1
+    n: T,           // last number we checked
+    nearest_six: T, // closest value divisible by 6
 }
 
 impl<T: num::PrimInt> Iterator for PrimeSequence<T> {
     type Item = T;
 
     fn next(&mut self) -> Option<T> {
         fn inc<V: num::PrimInt>(seq: &mut PrimeSequence<V>) {
-            if seq.bottom {
-                seq.bottom = false;
+            if seq.n < seq.nearest_six {
+                seq.n = seq.n + V::from(2).unwrap();
             } else {
-                seq.bottom = true;
-                seq.n = seq.n + V::from(6).unwrap();
+                seq.nearest_six = seq.nearest_six + V::from(6).unwrap();
+                seq.n = seq.nearest_six - V::one();
             }
         }
+        fn is_prime<V: num::PrimInt>(primes: &Vec<V>, n: V) -> bool {
+            for &prime in primes {
+                if prime * prime > n {                      // Should always be exit here on a prime
+                    return true                             // Bertrand's postulate states that for every integer n there is a prime number between n and 2n
+                }                                           // since all primes are at least 2, if we are dealing with a prime number the last prime squared will be greater than the next prime
+                if n % prime == V::zero() {
+                    return false
+                }
+            }
+            true                                           // should be mathematically impossible to get here, but rust still wants something
```

Comment:
I had a `panic!` here at one point for effect, but it seemed unnecessary.

Reply:
can we no longer add two and three to the primes vec? They are dead weight since we increment by 6. If we do that this `true` will be needed.

Reply:
```suggestion
            true
```